simple-navmesh-constraint.js

```diff
@@ -21,7 +21,28 @@ AFRAME.registerComponent('simple-navmesh-constraint', {
       default: ''
     }
   },
-  
+
+  init: function () {
+    this.updateNavmeshEntities = this.updateNavmeshEntities.bind(this);
+  },
+
+  play: function () {
+    this.el.sceneEl.addEventListener('child-attached', this.updateNavmeshEntities);
+    this.el.sceneEl.addEventListener('child-detached', this.updateNavmeshEntities);
+  },
+
+  pause: function () {
+    this.el.sceneEl.removeEventListener('child-attached', this.updateNavmeshEntities);
+    this.el.sceneEl.removeEventListener('child-detached', this.updateNavmeshEntities);
+  },
+
+  updateNavmeshEntities: function (evt) {
+    // Don't bother updating if the entity is not relevant to us
+    if (evt.detail.el.matches(`${this.data.navmesh}, ${this.data.exclude}`)) {
+      this.update();
```

Comment:
The code in `this.update()` is creating at least 5 new arrays. When you have lots of entities that matches navmesh or exclude selector, created in a single tick, it may be heavy to call this again and again. I would set a dirty flag and in tick call update() if dirty is true, similar to the raycaster component. And in inspector mode, tick won't be called, but it will be called when you leave the inspector mode and properly update navmesh entities.

Reply:
This was addressed in #28